src/network/bridge.rs

```diff
@@ -225,7 +237,7 @@ impl driver::NetworkDriver for Bridge<'_> {
         };
 
         // if the network is internal block routing and do not setup firewall rules
-        if self.info.network.internal {
+        if self.info.network.internal && data.mode == BridgeMode::L3 {
```

Comment:
if we wrap this entire part in `if let BridgeMode::L3 = data.mode {` up until the setup_firewall call we would only need to match once

Reply:
good idea, done